lib/uniffi-zcash/src/zcash_primitives/sapling/keys/extended_full_viewing_key.rs

```diff
@@ -0,0 +1,88 @@
+use std::sync::Arc;
+
+use zcash_primitives::zip32::ExtendedFullViewingKey;
+
+use crate::{
+    ZcashChildIndex, ZcashDiversifiableFullViewingKey, ZcashDiversifierIndex,
+    ZcashDiversifierIndexAndPaymentAddress, ZcashError, ZcashPaymentAddress, ZcashResult,
+};
+
+impl From<ExtendedFullViewingKey> for ZcashExtendedFullViewingKey {
+    fn from(key: ExtendedFullViewingKey) -> Self {
+        ZcashExtendedFullViewingKey(key)
+    }
+}
+
+impl From<&ZcashExtendedFullViewingKey> for ExtendedFullViewingKey {
+    fn from(key: &ZcashExtendedFullViewingKey) -> Self {
+        key.0.clone()
+    }
+}
+
+pub struct ZcashExtendedFullViewingKey(ExtendedFullViewingKey);
+
+impl ZcashExtendedFullViewingKey {
+    pub fn from_bytes(bytes: Vec<u8>) -> ZcashResult<Self> {
+        ExtendedFullViewingKey::read(bytes.as_slice())
+            .map(From::from)
+            .map_err(|error| error.to_string().into())
+    }
+
+    pub fn to_bytes(&self) -> Vec<u8> {
+        let mut bytes = vec![];
+        self.0.write(&mut bytes).unwrap();
+        bytes
+    }
+
+    pub fn derive_child(
+        &self,
+        i: ZcashChildIndex,
+    ) -> ZcashResult<Arc<ZcashExtendedFullViewingKey>> {
+        self.0
+            .derive_child(i.into())
+            .map_err(|_| ZcashError::Unknown)
```

Comment:
Similar here regarding the user friendly error. Maybe we should remove that error type (not in this PR) ? My reasoning:

When getting this error ("unknown error occurred") in a scoped environment, its easy to figure out what is happening. But in a bigger system (which just returns the error to upper layers), and that logs to files  or terminals, probably its not very orientative. What do you think ?